### PR TITLE
Feature/target

### PR DESCRIPTION
**Pull request by peter7z**
## Description
Targets can be created and then displayed on the map, next to already created targets.
Toasts system is used to display error and success messages.

### Screenshots
Targets: 
screen shot 2018-10-08 at 3 27 51 pm

Toasts:
screen shot 2018-10-08 at 3 28 10 pm

### CODE REVIEW

File: src/components/common/Map.js [hunk 1]
```diff
@@ -16,18 +16,38 @@ const Map = ({ handleClick, isMarkerShown, coords: { latitude, longitude }, targ
...
+        />
+      </div>
     }
+    {targets.map(({ target: { lat, lng, radius, id } }) => (
+      <div key={id} >
+        <Circle
+          options={{
+            strokeOpacity: '0',
+            fillOpacity: '.7',
+            fillColor: '#efc638'
```
Comment (xlasserre) [inline src/components/common/Map.js]: Could you store this color as a constant?

File: src/containers/HomePage.js [hunk 2]
```diff
@@ -14,30 +16,48 @@ class HomePage extends React.Component {
...
     this.state = {
       coords: null,
-      target: null
+      target: null,
     };
   }
 
   componentDidMount() {
     this.props.getTopics();
+    this.props.getTargets();
```
Comment (xlasserre) [inline src/containers/HomePage.js]: You could deconstruct ```this.props``` here to retrieve the two functions.

**Review by xlasserre — CHANGES_REQUESTED** [review]
Nice job! I only added two small comments.

File: src/components/user/TargetForm.js [hunk 3]
```diff
@@ -77,7 +79,7 @@ let TargetForm = ({ error, handleSubmit, submitting, intl, topics, eraseTarget }
                 classNamePrefix="react-select"
                 component={SelectBox}
                 label={intl.formatMessage(messages.topic)}
-                name="topic"
+                name="topic_id"
```
Comment (agch-dev) [inline src/components/user/TargetForm.js]: why are you using snake_case? the apiService already transforms it from camelCase to snake_case before sending it to the backend.

File: src/containers/HomePage.js [hunk 4]
```diff
@@ -46,8 +66,9 @@ class HomePage extends React.Component {
             containerElement={<div className="maps-container" />}
             mapElement={<div className="google-maps" />}
             handleClick={this.mapClick}
-            target={!this.props.match.isExact ? this.state.target : null}
+            target={(this.props.location.pathname == '/create-target') ? this.state.target : null}
```
Comment (agch-dev) [inline src/containers/HomePage.js]: use destructuring for both props and state

**Review by agch-dev — CHANGES_REQUESTED** [review]
Nice work! Just a few comments

File: src/components/user/TargetForm.js (hunk 3 above)
Comment (peter7z) [reply src/components/user/TargetForm.js]: True! Forgot about that.

File: src/constants/constants.js [hunk 5]
```diff
@@ -11,3 +11,5 @@ export const gender = [
   { value: 'female', id: 'gender.female' },
   { value: 'other', id: 'gender.other' }
 ];
+
+export const macCheeseColor = '#efc638';
```
Comment (xlasserre) [inline src/constants/constants.js]: I think for this you should have a different file called something like styleConstants.js, otherwise it's kinda messy.

**Review by xlasserre — CHANGES_REQUESTED** [review]
Just left one small comment regarding the constants file. The rest is good!

File: src/containers/HomePage.js [hunk 6]
```diff
@@ -46,8 +69,9 @@ class HomePage extends React.Component {
             containerElement={<div className="maps-container" />}
             mapElement={<div className="google-maps" />}
             handleClick={this.mapClick}
-            target={!this.props.match.isExact ? this.state.target : null}
-            coords={this.state.coords}
+            target={(location.pathname == '/create-target') ? target : null}
```
Comment (felipegarcia92) [inline src/containers/HomePage.js]: You have this route in a routesPaths file, right? Can you compare to that constant instead of typing it by hand?

**Review by felipegarcia92 — APPROVED** [review]

**Review by xlasserre — APPROVED** [review]

**Pull request merged**